local-interchaintest/src/helpers/covenant.rs

```diff
@@ -0,0 +1,92 @@
+use localic_std::{modules::cosmwasm::contract_query, transactions::ChainRequestBuilder};
+
+pub enum Covenant<'a> {
+    TwoPartyPol {
```

Comment:
do you think there's a way to avoid these enum variants? it is somewhat limiting to box covenant workflows into specific types, tho we don't really have the generalized covenants yet so it may be too early to think about this.

Reply:
I'm doing this for the queries basically since both covenants have the same queries, I don't want to duplicate any code and I want to use the right message of each covenant API so that if we ever modify one of them, it will ask us to change it here too. I found this is the cleanest way to do it but if you have any other ideas we can discuss

Reply:
nop that makes sense to me! we should normalize the queries in the future..